pkg/patterns/declarative/pkg/applier/applylib.go

```diff
@@ -43,26 +54,34 @@ func (a *ApplySetApplier) Apply(ctx context.Context, opt ApplierOptions) error {
 	}
 
 	restMapper := opt.RESTMapper
+	tooling := a.Tooling
+	if tooling == "" {
+		tooling = opt.ParentRef.GroupVersionKind().Kind
+	}
 
 	options := applyset.Options{
-		PatchOptions: patchOptions,
-		RESTMapper:   restMapper,
-		Client:       dynamicClient,
+		Parent:        opt.ParentRef,
+		PatchOptions:  patchOptions,
+		DeleteOptions: a.deleteOptions,
+		RESTMapper:    restMapper,
+		Client:        dynamicClient,
+		Prune:         opt.Prune,
+		Tooling:       tooling,
 	}
 	s, err := applyset.New(options)
 	if err != nil {
 		return fmt.Errorf("error creating applyset: %w", err)
 	}
 
-	// Populate the namespace on any namespace-scoped objects
-	if opt.Namespace != "" {
-		for _, obj := range opt.Objects {
-			gvk := obj.GroupVersionKind()
```

Comment:
Lol - looks like I assumed restMapper cached as well ;-)

Any rason to move this out of the loop?  Are you trying to catch errors where the GVK is not yet known?  I don't _think_ you want to error out here, I think the correct answer is probably a partial apply (with correct management of the resources-in-use annotation in the parent)